auto_rx/autorx/aprs.py

```diff
@@ -362,7 +360,8 @@ def __init__(
         self.aprsis_reconnect = aprsis_reconnect
         self.upload_timeout = upload_timeout
         self.upload_queue_size = upload_queue_size
-        self.synchronous_upload_time = synchronous_upload_time
+        self.upload_time = upload_time
+        self.next_upload = datetime.datetime.now() + datetime.timedelta(seconds=upload_time)
```

Comment:
Using datetime.datetime.now() uses wall-clock time. When clock jumps, this code is confused. For example, a Raspberry Pi does not have a real-time clock, software starts up, then NTP synchronizes and the clock jumps.  It's better to use a monotonic clock which is not affected by wall-clock changes (Python 3 required, but what doesn't these days):

```
self.next_upload = time.monotonic() + upload_time

...

if time.monotonic() > self.next_upload:
```

Reply:
good catch :)